### PR TITLE
py reader queue support test mode for speed test

### PR DESCRIPTION
**Pull request by jacquesqiao**
当处理数据的速度变得非常快的时候，py_reader也常常跟不上数据输入的速度，表现为queue经常为空，这是由于py_reader在python端只有一个python线程来往queue里灌数据，速度不够快，在测试加速比的时候，可以先用这种模式来排除reader的影响(因为可以通过重写c++ reader的方式来加速数据读取的速度)。

在某个单机多线程的case下可以看到提速50%，能测出排除数据干扰的加速比，

### CODE REVIEW

File: paddle/fluid/operators/reader/blocking_queue.h [hunk 1]
```diff
@@ -72,7 +72,9 @@ class BlockingQueue {
     if (!queue_.empty()) {
       PADDLE_ENFORCE_NOT_NULL(elem);
       *elem = queue_.front();
-      queue_.pop_front();
+      if (!test_mode_) {
```
Comment (typhoonzero) [inline paddle/fluid/operators/reader/blocking_queue.h]: use `if (UNLIKELY(!test_mode_))` to speed up normal case?

**Comment by typhoonzero** [comment]
I think this is a little bit tricky, in order to figure out data reader IO is the current bottleneck or not, we'd better output the IO time per batch and the IO percentage of the total batch time.

A test mode argument may introduce something hard to understand for user API.

File: paddle/fluid/operators/reader/blocking_queue.h [hunk 2]
```diff
@@ -31,8 +31,8 @@ class BlockingQueue {
   // is a workaround and a simplified version of framework::Channel as it
   // doesn't support GPU and it implements on buffered blocking queue.
  public:
-  explicit BlockingQueue(size_t capacity)
-      : capacity_(capacity), closed_(false) {
+  explicit BlockingQueue(size_t capacity, bool test_mode = false)
+      : capacity_(capacity), test_mode_(test_mode), closed_(false) {
```
Comment (chengduoZH) [inline paddle/fluid/operators/reader/blocking_queue.h]: `test_mode` is a bit confuse, maybe some user think it is for inference.

**Comment by jacquesqiao** [comment]
Use gflags to set reader_queue_speed_test_mode flag, do not expose argument to user.
fix with https://github.com/PaddlePaddle/Paddle/pull/13864

**Pull request closed without merging**